Mathlib/LinearAlgebra/FreeModule/Finite/Rank.lean

```diff
@@ -436,7 +436,98 @@ instance Module.Finite.span_singleton (x : V) : Module.Finite R (R ∙ x) :=
 instance Module.Finite.span_finset (s : Finset V) : Module.Finite R (span R (s : Set V)) :=
   ⟨(Submodule.fg_top _).mpr ⟨s, rfl⟩⟩
 
-variable {K R}
+variable {K R M}
+variable [AddCommGroup M] [Module R M] [Module.Finite R M]
+
+lemma LinearIndependent.finset_toSet (s : Finset M) : LinearIndependent R ((↑) : (↑s : Set M) → M) ↔
+    LinearIndependent R ((↑) : s → M) := Iff.rfl
+
+lemma FiniteDimensional.finrank_add_finrank_quotient_le (N : Submodule R M) :
+    finrank R N + finrank R (M ⧸ N) ≤ finrank R M := by
```

Comment:
Just letting you know: I'm trying to prove the `rank` version of this and I've established a [key lemma](https://github.com/leanprover-community/mathlib4/compare/master...rank_quotient_add_or_prod#diff-23b5582f59f20ba2c8878d54747dac6dcabb32114e16e5f5d92dd53652cabe5eR854-R857) in another branch using your prior work in #8842.